source/Generic/DateImporter/DateImporter.psm1

```diff
@@ -846,5 +883,123 @@ function Get-SteamLicenses
         }
         $LicensesList.Add($License)
     }
+    return $LicensesList
+}
+
+function Export-OriginLicenses
+{
+    param(
+        $scriptMainMenuItemActionArgs
+    )
+    
+    $libraryName = "Origin"
+    
+    $LicensesList = Get-OriginLicenses $libraryName
+    if ($LicensesList.count -eq 0)
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_LicensesNotFoundMessage"), "$libraryName Date Importer")
+        return
+    }
+    Export-Licenses $libraryName $LicensesList
+}
+
+function Add-OriginDates
+{
+    param (
+        $gameDatabase
+    )
+
+    if ($gameDatabase.count -eq 0)
+    {
+        return
+    }
+
+    $libraryName = "Origin"
+    $LicensesList = Get-OriginLicenses $libraryName
+    if ($LicensesList.count -eq 0)
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_LicensesNotFoundMessage"), "$libraryName Date Importer")
+        return
+    }
+
+    Set-DatesFromLicenses $libraryName $LicensesList $gameDatabase
+}
+
+function Get-OriginLicenses
+{
+    param (
+        $libraryName
+    )
+    
+    [System.Collections.Generic.List[object]]$LicensesList = @()
+
+    $authUrl = "https://accounts.ea.com/connect/auth?client_id=ORIGIN_JS_SDK&response_type=token&redirect_uri=nucleus:rest&prompt=none"
+    $loginStatus = Get-LoginStatusViaJson $authUrl
+    if ($loginStatus -eq $false)
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_UsetNotLoggedMessage") -f $libraryName), "$libraryName Date Importer")
+        return $LicensesList
+    }
+
+    $webView = $PlayniteApi.WebViews.CreateOffscreenView()
+    $webView.NavigateAndWait($authUrl)
+    $pageSource = $webView.GetPageSource()
+    $webView.Dispose()
+
+    $jsonValid = Get-IsJsonValidFromPage $pageSource
+
+    if($jsonValid -eq $false) 
```

Comment:
can you add a space after the `if` here please?

Reply:
Done in 559566c951bf00f312b54bf5f8efed31753055ee. 
Would you happen to have a config file for your IDE with your formatting preferences? So i don't need to get on your nerves with that stuff 😄

Reply:
I've not looked into settting a config file to be honest, I'll look in the weekend how to do it.